format/formatter.go

```diff
@@ -47,6 +47,132 @@ type Formatter interface {
 	Close() error
 }
 
+type anyRecord struct {
+	v interface{}
+}
+
+func (r *anyRecord) UnmarshalJSON(b []byte) error {
+	var i interface{}
+	err := json.Unmarshal(b, &i)
+	if err != nil {
+		return err
+	}
+
+	switch j := i.(type) {
+	case map[string]interface{}:
+		if d, ok := r.formatDate(j); ok {
+			r.v = d
+			return nil
+		}
+
+		r.traverseMap(&j)
+		r.v = j
+	default:
+		r.v = i
+	}
+
+	return err
+}
```

Comment:
I don't think I fully comprehend what is happening here, mostly because I am new to `go`, perhaps we can discuss this part IRL? In addition, perhaps we should add some context to the errors that we return?

Reply:
The documentation and code was updated, now it is easier to understand what this function does